lib/OrdinaryDiffEqDifferentiation/src/alg_utils.jl

```diff
@@ -16,16 +16,14 @@ function _alg_autodiff(alg::Union{OrdinaryDiffEqExponentialAlgorithm{CS, AD},
     alg.autodiff
 end
 
+# Type-stable helper functions using dispatch instead of runtime conditionals
+_unwrap_autodiff(::Val{true}) = AutoForwardDiff()
+_unwrap_autodiff(::Val{false}) = AutoFiniteDiff()
+_unwrap_autodiff(x) = x
+
 function alg_autodiff(alg)
     autodiff = _alg_autodiff(alg)
-
-    if autodiff == Val(true)
-        return AutoForwardDiff()
-    elseif autodiff == Val(false)
-        return AutoFiniteDiff()
-    else
-        return autodiff
-    end
```

Comment:
alternatively, this could be
```
    if autodiff isa Val{true}
        return AutoForwardDiff()
    elseif autodiff isa Val{false}
        return AutoFiniteDiff()
    else
        return autodiff
    end
```